client/app/user/profile/profile.controller.js

```diff
@@ -9,23 +9,117 @@ angular.module('observatory3App')
         loggedInUser = user;
         $scope.isuser = loggedInUser._id === $stateParams.id;
         $scope.loggedInUserRole = loggedInUser.role;
-      });
 
-      $http.get('/api/users/' + $stateParams.id).success(function(user){
-          $scope.user = user;
-          $scope.originalRole = user.role;
-          $http.get('/api/commits/user/' + user.githubProfile).success(function(commits){
-              $scope.user.commits = commits;
-          });
-          // get all users projects information
-          $scope.projects = [];
-          user.projects.forEach(function(projectId){
-            $http.get("/api/projects/" + projectId).success(function(project){
-              $scope.projects.push(project);
+        //If you can access private data
+        if ($scope.isuser || $scope.loggedInUserRole === 'admin'){
+            $http.get('/api/users/' + $stateParams.id+'/private').success(function(user){
+                $scope.user = user;
+                if (!$scope.user.bio){
+                    $scope.user.bio = "An awesome RCOS developer!";
+                }
+                $scope.originalRole = user.role;
+                //Total attendance is current attendance in large and small groups
+                var u = $scope.user;
+                u.full={normal:0, bonus:0};
+                u.small={normal:0, bonus:0};
+                u.unverified = 0;
+                for (var b = 0; b < u.attendance.length; b++){
```

Comment:
this is repeat code from the admin page. Any way we could modularize this? Perhaps using a service that processes a users data fields